ui/src/data_explorer/components/WriteDataForm.js

```diff
@@ -0,0 +1,112 @@
+import React, {Component, PropTypes} from 'react'
+import {connect} from 'react-redux'
+import {bindActionCreators} from 'redux'
+
+import DatabaseDropdown from 'shared/components/DatabaseDropdown'
+
+import {writeData} from 'src/data_explorer/apis'
+import {publishAutoDismissingNotification} from 'shared/dispatchers'
+import {errorThrown as errorThrownAction} from 'shared/actions/errors'
+
+class WriteDataForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedDatabase: null,
+    }
+
+    this.handleSelectDatabase = ::this.handleSelectDatabase
+    this.handleError = ::this.handleError
+    this.handleWrite = ::this.handleWrite
+  }
+
+  handleSelectDatabase(item) {
+    this.setState({selectedDatabase: item.text})
+  }
+
+  handleError(error) {
+    const {errorThrown} = this.props
+    errorThrown(error)
+  }
+
+  async handleWrite() {
```

Comment:
As much as possible, my opinion is that we want to handle AJAX an errors in the thunk (the async action creators), not the component. Because separation of concerns. Props for error handling, and cool to see your use of `errorThrown as errorThrownAction` in `mapDispatchToProps`, but `async handleWrite()` should just be `handleWrite()` and should just trigger the thunk, i.e. `this.props.writeData(source, selectedDatabase, influxQLCommand)`, and as mentioned the AJAX and error handling should be handled in that thunk.